x-pack/metricbeat/module/azure/mock_service.go

```diff
@@ -43,7 +43,7 @@ func (client *MockService) GetMetricNamespaces(resourceId string) (armmonitor.Me
 
 // GetMetricValues is a mock function for the azure service
 func (client *MockService) GetMetricValues(resourceId string, namespace string, timegrain string, timespan string, metricNames []string, aggregations string, filter string) ([]armmonitor.Metric, string, error) {
-	args := client.Called(resourceId, namespace)
+	args := client.Called(resourceId, namespace, timegrain, timespan, metricNames, aggregations, filter)
```

Comment:
You can just add some description what each parameter is here used for
Maybe you can update the  GetMetricValues description

Reply:
I noticed there is an unused method in the interface. I'll update the docs for the other methods and remove the unused ones in a cleanup PR.